File: backend/api/schemas/patient_verification.py
from pydantic import BaseModel, Field


class PatientVerificationOtpRequest(BaseModel):
    draft_id: str = Field(min_length=1)
    method: str = Field(min_length=1)
    identifier: str = Field(min_length=1)


class PatientVerificationOtpResponse(BaseModel):
    challenge_id: str
    masked_destination: str
    expires_in_seconds: int
    demo_otp: str | None = None


class PatientVerificationOtpVerifyRequest(BaseModel):
    draft_id: str = Field(min_length=1)
    challenge_id: str = Field(min_length=1)
    otp: str = Field(min_length=1)


class PatientVerificationOtpVerifyResponse(BaseModel):
    verification_token: str
    status: str
    expires_in_seconds: int

File: patient-web/src/api/patientVerification.ts
import { apiRequest } from "./client";

export interface VerificationOtpChallenge {
  challenge_id: string;
  masked_destination: string;
  expires_in_seconds: number;
  demo_otp: string | null;
}

export interface PatientVerificationResult {
  verification_token: string;
  status: "VERIFIED";
  expires_in_seconds: number;
}

export async function requestPatientVerificationOtp(
  draftId: string,
  method: "ABHA" | "AADHAAR",
  identifier: string,
): Promise<VerificationOtpChallenge> {
  const response = await apiRequest<{
    data: VerificationOtpChallenge;
  }>("/identity/otp", {
    method: "POST",
    body: JSON.stringify({
      draft_id: draftId,
      method,
      identifier,
    }),
  });

  return response.data;
}

export async function verifyPatientIdentityOtp(
  draftId: string,
  challengeId: string,
  otp: string,
): Promise<PatientVerificationResult> {
  const response = await apiRequest<{
    data: PatientVerificationResult;
  }>("/identity/otp/verify", {
    method: "POST",
    body: JSON.stringify({
      draft_id: draftId,
      challenge_id: challengeId,
      otp,
    }),
  });

  return response.data;
}
